src/tasks/service/tasks.service.ts

```diff
@@ -32,6 +32,20 @@ export class TasksService {
     }
     return await this.taskModel.find({ managerId }).exec();
   }
+  async getAllTasksForEmployee(
+    employeeId: string,
+    businessId: string,
+  ): Promise<Task[]> {
+    if (!employeeId || !businessId) {
+      throw new BadRequestException('employeeID and businessID is required');
+    }
+    const tasksByBusiness = await this.taskModel.find({ businessId }).exec();
+    const tasksByBusinessByEmployee = tasksByBusiness.filter((task) =>
+      task.employee.includes(new Types.ObjectId(employeeId)),
```

Comment:
אי אפשר להכניס את זה לשאילתא?
https://stackoverflow.com/questions/18148166/find-document-with-array-that-contains-a-specific-value

Reply:
איזו שאילתא ניסית ולא עבדה לך?

Reply:
ניסיתי את שלושתן:
PersonModel.find({ favoriteFood : { $all : ["sushi"] }, ...})

PersonModel.find({ favouriteFoods: { "$in" : ["sushi"]} }, ...);

PersonModel.find({ favouriteFoods: "sushi" }, ...); // favouriteFoods contains "sushi"
על כל אובייקט אני בעצם צריכה לחפש בתוך המערך של  employee
{
  "businessId": "company123",
  "taskName": "Develop new feature",
  "managerId": "employee123",
  "description": "Develop and implement the new authentication feature",
  "targetDate": "2024-12-31",
  "employee": [
    "664cba7ee786ab5c121aa40b",
    "123765434567hgfdfghjkhgfgh"
  ],
  "urgency": 1,
  "status": 1,
  "completionDate": "2024-12-31",
  "directLink": "http://localhost:3001/api#/Workers/WorkersController_create"
}

Reply:
PersonModel.find({ employee: "sushi" }, ...); // favouriteFoods contains "sushi"
זאת הצורה שאני מכירה, מה חזר לך מהמונגו?

Reply:
?

Reply:
אני לא מצליחה להשתמש בזה זה מחזיר לי  מערך ריק..

.

Reply:
האם את יכולה לשלוף משימות לפי employee בתוך mongo compass?